pages/api/users/[username]/index.js

```diff
@@ -14,12 +17,17 @@ export default async function handler(req, res) {
       .status(400)
       .json({ error: "Invalid request: GET request required" });
   }
-
-  const { status, profile } = await getUserApi(req.query.username);
+  
+  const { status, profile } = await getUserApi(req, res, req.query.username);
   return res.status(status).json(profile);
 }
 
-export async function getUserApi(username) {
+export async function getUserApi(req, res, username) {
+  let sameUser = false;
+  const session = await unstable_getServerSession(req, res, authOptions);
+  if (session) {
+    sameUser = (session.username === username) ? true : false;
+  }
```

Comment:
I think as it is already false the extra check is not needed

```suggestion
  if (session && session.username === username) {
    sameUser = true;
  }
```

I will fix via a temporary branch